hooks/pre-commit

```diff
@@ -1,25 +1,29 @@
-#!/bin/bash
-if clang-format --version | grep -q 'version 11\.'; then
-   CLANG_FORMAT_EXECUTABLE="clang-format"
-else
-   CLANG_FORMAT_EXECUTABLE="clang-format-11"
-fi
+#!/bin/sh
+
+minVersion="14.0.0"
+
+name=""
+version=""
+
+for file in $(find $(echo "$PATH" | tr ':' ' ') -maxdepth 1 -type f -executable -name 'git-clang-format*'); do
+  curName="$(basename "$file" | sed 's/^git-//')"
+  curVersion="$("$curName" --version | awk '{print $3}')"
 
-if ! command -v $CLANG_FORMAT_EXECUTABLE &> /dev/null
-then
-    echo $CLANG_FORMAT_EXECUTABLE does not exist, make sure to install it
-    exit 1
+  if [ "$(printf '%s\n' "$curVersion" "$version" "$minVersion" | sort -V | tail -n 1)" = "$curVersion" ]; then
+    name="$curName"
+    version="$curVersion"
+  fi
+done
+
+if [ -z "$name" ]; then
+  echo "Could not find a suitable clang-format installation. Install clang-format with at least version $minVersion"
+  exit 1
```

Comment:
You should store the entire found `git-clang-format` script name, and the error message should say it couldn't find a suitable `git-clang-format` script.

Reply:
I don't agree that it should store the whole `git-clang-format` script name, because I think it should store the name of the suitable clang-format installation. That name is  used to get the version, and then used to for the `git clang-format` command. I will add a note to the error message that the clang-format installation needs to include the `git-clang-format` script.